MIGRATION.ru.md

```diff
@@ -1,5 +1,743 @@
+на удаление этого модификатора. Приходилось делать две декларации.
+
+Было:
+
+```js
+// my-dom-block_my-mod_my-val.js
+
+modules.define('my-dom-block', function(provide, MyDomBlock) {
+
+MyDomBlock.decl({
+    onSetMod : {
+        'my-mod' : {
+            '' : function() { /* ... */ } // декларация для удаления модификатора _my-mod_my-val
+        }
+    }
+
+});
+
+provide(MyDomBlock.decl({ modName : 'my-mod', modVal : 'my-val' }, { /* ... */ }));
+
+});
+```
+
+Стало:
+
+```js
+modules.define('my-dom-block', function(provide, MyDomBlock) {
+
+provide(MyDomBlock.declMod({ modName : 'my-mod', modVal : 'my-val' }, {
+    onSetMod : {
+        'mod1' : {
+            '' : function() { /* ... */ } // декларация для удаления модификатора _my-mod_my-val
+        }
+    }
+}));
+```
+
+Задача: [#1025](https://github.com/bem/bem-core/issues/1025).
+
+Появился сокращённый синтаксис для декларации поведения на изменение модификатора.
+
+Было:
+
+```js
+onSetMod : {
+    'my-mod' : {
+        '*' : function(modName, modVal, prevModVal) {
+            if(prevModVal === 'my-val') {
+                /* ... */ // декларация для изменения _my-mod_my-val в любое другое значение
+            }
+        }
+    }
+}
+```
+
+Стало:
+
+```js
+onSetMod : {
+    'my-mod' : {
+        '~my-val' : function() { /* ... */ } // декларация для изменения значения my-mod из my-val в любое другое значение
+        }
+    }
+}
+```
+
+Было:
+
+```js
+onSetMod : {
+    'my-mod' : {
+        '*' : function(modName, modVal) {
+            if(modVal !== 'my-val') {
+                /* ... */ // декларация для изменения my-mod в любое значение, кроме my-val
+            }
+        }
+    }
+}
+```
+
+Стало:
+
+```js
+onSetMod : {
+    'my-mod' : {
+        '!my-val' : function() { /* ... */ } // декларация для изменения my-mod в любое значение, кроме my-val
+        }
+    }
+}
+```
+
+Задача: [#1072](https://github.com/bem/bem-core/issues/1072).
+
+
+### Ленивая инициализация
+
+Функциональность поля `live` была разделена на две части: поле `lazyInit` и метод `onInit()`.
+
+Было:
+
+```js
+modules.define('my-dom-block', ['i-bem__dom'], function(provide, BEMDOM) {
+
+provide(BEMDOM.decl(this.name, { /* ... */ }, {
+    live : true
+}));
+
+});
+```
+
+Стало:
+
+```js
+modules.define('my-dom-block', ['i-bem-dom'], function(provide, bemDom) {
+
+provide(bemDom.declBlock(this.name, { /* ... */ }, {
+    lazyInit : true
+}));
+
+});
+```
+
+Было:
+
+```js
+modules.define('my-dom-block', ['i-bem__dom'], function(provide, BEMDOM) {
+
+provide(BEMDOM.decl(this.name, { /* ... */ }, {
+    live : function() {
+        /* ... */
+    }
+}));
+
+});
+```
+
+Стало:
+
+```js
+modules.define('my-dom-block', ['i-bem-dom'], function(provide, bemDom) {
+
+provide(bemDom.declBlock(this.name, { /* ... */ }, {
+    lazyInit : true,
+
+    onInit : function() {
+        /* ... */
+    }
+}));
+
+});
+```
+
+Было:
+
+```js
+modules.define('my-dom-block', ['i-bem__dom'], function(provide, BEMDOM) {
```

Comment:
Please provide information about the way to work with `live` in BEMJSON now.

Before:
```js
{
    block: 'b1',
    js: { live: false }
}
```

Now: ???